src/contexts/Photos/Photos.context.tsx

```diff
@@ -131,12 +145,37 @@ export const PhotosContextProvider: React.FC = ({ children }) => {
     }),
   );
 
+  useEffect(() => {
+    checkShouldEnableSync().catch((error) => {
+      reportError(error);
+    });
+  }, []);
+
   useEffect(() => {
     if (syncStatus === PhotosSyncStatus.Completed) {
       setUploadingPhotosItem(null);
     }
   }, [syncStatus]);
 
+  async function checkShouldEnableSync() {
+    const permissionStatus = await permissions.getPermissionsStatus();
+
+    // No permissions, disable the sync
+    if (
+      permissionStatus === MediaLibrary.PermissionStatus.UNDETERMINED ||
+      permissionStatus === MediaLibrary.PermissionStatus.DENIED
+    ) {
+      setSyncEnabled(false);
+
+      return false;
+    }
+
+    const isEnabled = (await asyncStorageService.getItem(AsyncStorageKey.PhotosSyncEnabled)) === 'true';
```

Comment:
Any change of the format of how the asyncStorageService stores if the photos sync is enabled or not would affect all the code that checks against a string. I suggest hiding this check behind a method from the asyncStorageService like:

`asyncStorageService.isPhotosSyncEnabled() -> boolean`

Thus you have more flexibility to change the format, the key name or anything else in the future.